pkg/install/install_test.go

```diff
@@ -120,3 +122,22 @@ func TestDeployARMTemplate(t *testing.T) {
 		})
 	}
 }
+
+func TestResourceProviderVersion(t *testing.T) {
+	ctx := context.Background()
+
+	i := &Installer{
+		doc: &api.OpenShiftClusterDocument{
+			OpenShiftCluster: &api.OpenShiftCluster{
+				Properties: api.OpenShiftClusterProperties{},
+			},
+		},
+	}
+
+	i.addResourceProviderVersion(ctx)
```

Comment:
check the error returned

Reply:
there's a missing step here with a mock DB client where you check that PatchWithLease is called

Reply:
Done and done